tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -113,44 +119,81 @@ private async void GetIcons()
         }
     }
 
-    internal string CreateFullCommandLine(int? pid, HWND? hwnd)
-    {
-        return "\"" + Executable + "\"" + CreateCommandLine(pid, hwnd);
-    }
-
-    internal string CreateCommandLine(int? pid, HWND? hwnd)
-    {
-        var commandLine = $" {OtherArgs}";
-
-        if (ArgType == ExternalToolArgType.Hwnd && hwnd is not null)
-        {
-            commandLine = $" {ArgPrefix} {hwnd:D} {OtherArgs}";
-        }
-        else if (ArgType == ExternalToolArgType.ProcessId && pid is not null)
-        {
-            commandLine = $" {ArgPrefix} {pid:D} {OtherArgs}";
-        }
-
-        return commandLine;
-    }
+    internal async Task<bool> Invoke(int? pid, HWND? hwnd)
+    {
+        var result = false;
 
-    internal virtual Process? Invoke(int? pid, HWND? hwnd)
-    {
         try
-        {
-            var toolProcess = new Process();
-            toolProcess.StartInfo.FileName = Executable;
-            toolProcess.StartInfo.Arguments = CreateCommandLine(pid, hwnd);
-            toolProcess.StartInfo.UseShellExecute = false;
-            toolProcess.StartInfo.RedirectStandardOutput = true;
```

Comment:
>RedirectStandardOutput [](http://example.com/codeflow?start=34&length=22)

Was it intentional you removed this? This would prevent us from capturing the standard output of the tool we launched.

Reply:
Oops, nope that was a mistake. Fixed now, thanks.